force-app/main/default/classes/Addresses.cls

```diff
@@ -165,210 +166,293 @@ public inherited sharing class Addresses extends fflib_SObjects {
+        }
+        if (contactsToUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)contactsToUpdate);
         }
 
-        // get all Contacts by HH, that might need updating
-        // note that if we have an organizational account, our query will not pick up any contacts (by design!)
-        Map<Id, List<Contact>> mapAccIdListCon = new Map<Id, List<Contact>>();
-        Map<Id, Account> accountsById = getAccountsByIds(mapAccIdAddr.keySet());
-        Set<Id> setHHId =  mapAccIdAddr.keySet();
-
-        for (Contact con : contactSelector.getContactAddressFieldsForContactsIn(setHHId)) {
-            List<Contact> listConHH = mapAccIdListCon.get(con.AccountId);
-            if (listConHH == null) {
-                listConHH = new List<Contact>();
-                mapAccIdListCon.put(con.AccountId, listConHH);
+        return householdInfoByAccountId;
+    }
+
+    private static Map<Id, List<Contact>> getContactsByHouseholdAccountId(Set<Id> accountIds) {
+        Map<Id, List<Contact>> contactsByHouseholdAccountId = new Map<Id, List<Contact>>();
+
+        for (Contact contact : contactSelector.getContactAddressFieldsForContactAccountsIn(accountIds)) {
+            List<Contact> householdContacts = contactsByHouseholdAccountId.get(contact.AccountId);
+            if (householdContacts == null) {
+                householdContacts = new List<Contact>();
+                contactsByHouseholdAccountId.put(contact.AccountId, householdContacts);
             }
-            listConHH.add(con);
+            householdContacts.add(contact);
         }
+        return contactsByHouseholdAccountId;
+    }
 
-        // lists of objects to update
-        List<Account> listAccUpdate = new List<Account>();
-        List<Address__c> listAddrUpdate = new List<Address__c>();
-        List<Contact> listConUpdate = new List<Contact>();
+    private static NPSP_Address getAddressToSyncToAccountsAndContacts(Id accountId,
+            Map<Id, HouseholdInfo> householdInfoByAccountId) {
 
-        // now go through each HH, and do all the work!
-        for (Id accId : listAccId) {
-
-            // pick the address to use
-            HHInfo hhinfo = mapAccIdHHInfo.get(accId);
-            Address__c addrDefault = hhinfo.addrDefault;
-            Address__c addr = hhinfo.addrSeasonal;
-            if (addr == null) {
-                addr = addrDefault;
-            }
+        HouseholdInfo householdInfo = householdInfoByAccountId.get(accountId);
 
-            // update HH Account's Billing address fields
-            NPSP_Address npspAddress = new NPSP_Address(addr);
+        return new NPSP_Address(householdInfo.seasonalAddress != null ?
+               householdInfo.seasonalAddress : householdInfo.defaultAddress);
+    }
 
-            NPSP_HouseholdAccount npspHouseholdAccount = new NPSP_HouseholdAccount(accountsById.get(accId));
+    private static void updateAddressInfoForAccountsAndContacts(Set<Id> accountIds,
```

Comment:
This method feels like it has quite a few parameters. Are these able to be consolidated, or maybe a class that could be extracted?